modules/monitoring-example-service-endpoint-authentication-settings.adoc

```diff
@@ -0,0 +1,152 @@
+
+.Sample YAML authentication with a bearer token
+====
+The following sample shows bearer token settings for a `Secret` object named `example-bearer-auth` in the `ns1` namespace:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: example-bearer-auth
+  namespace: ns1
+stringData:
+  token: <authentication_token> #<1>
+----
+<1> Specify an authentication token.
+
+The following sample shows bearer token authentication settings for a `ServiceMonitor` CRD. The example uses a `Secret` object named `example-bearer-auth`:
+
+[source,yaml]
+----
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: prometheus-example-monitor
+  namespace: ns1 
+spec:
+  endpoints: 
+  - authorization:
+      credentials:
+        key: token #<1>
+        name: example-bearer-auth #<2>
+  selector: 
+    matchLabels:
+      app: prometheus-example-app
+----
+<1> The key that contains the authentication token in the specified `Secret` object.
+<2> The name of the `Secret` object that contains the authentication credentials.
+
+[IMPORTANT]
+=====
+Use an `authorization` object instead of `bearerTokenFile` to configure bearer token. 
+If you use the `bearerTokenFile` configuration, the `ServiceMonitor` resource is rejected.
+=====
+====
+
+.Sample YAML for Basic authentication
+====
+The following sample shows Basic authentication settings for a `Secret` object named `example-basic-auth` in the `ns1` namespace:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: example-basic-auth
+  namespace: ns1
+stringData:
+  user: <basic_username> #<1>
+  password: <basic_password>  #<2>
+----
+<1> Specify a username for authentication.
+<2> Specify a password for authentication.
+
+The following sample shows Basic authentication settings for a `ServiceMonitor` CRD. The example uses a `Secret` object named `example-basic-auth`:
+
+[source,yaml]
+----
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: prometheus-example-monitor
+  namespace: ns1 
+spec:
+  endpoints: 
+  - basicAuth:
+      username:
+        key: user #<1>
+        name: example-basic-auth #<2>
+      password:
+        key: password #<3>
+        name: example-basic-auth #<2>
+  selector: 
+    matchLabels:
+      app: prometheus-example-app
+----
+<1> The key that contains the username in the specified `Secret` object.
+<2> The name of the `Secret` object that contains the Basic authentication.
+<3> The key that contains the password in the specified `Secret` object.
+====
+
+.Sample YAML authentication with OAuth 2.0
+====
+The following sample shows OAuth 2.0  settings for a `Secret` object named `example-oauth2` in the `ns1` namespace:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: example-oauth2
+  namespace: ns1
+stringData:
+  id: <oauth2_id> #<1>
+  secret: <oauth2_secret> #<2>
+  token: <oauth2_authentication_token> #<3>
```

Comment:
not needed?

Reply:
Hello @simonpasquier what value exactly is not needed here? Is it the token one? Is it because we specify the tokenURL in the Servicemonitor object?

Also, I used the [remote-write](https://docs.openshift.com/container-platform/4.15/observability/monitoring/configuring-the-monitoring-stack.html#example-remote-write-authentication-settings_configuring-the-monitoring-stack) section as an exmaple, is there an error too then? I
Thank you for clarification!

Reply:
yes the `token` field should be removed here, and also from the OAuth2 remote-write example.

Reply:
okay, thank you for clarification!